### ISSUE TITLE
Make the "WTF" configurable for primary-aged children

### ISSUE THREAD

**Issue by twistylittleworkshop** [mavoweb/mavo#588]
I am hoping to use a Mavo app with primary-aged children. I cannot find a UI-tweak to adjust the "WTF?!" button on the login authorisation dialogue.

Can this be easily tweaked, or will this necessitate a locally patched version of Mavo to achieve? If so, where does this code reside?

Many thanks.

**Comment by LeaVerou** [comment]
This is leftover from the early days, and it has just not been changed because nobody ever asked for it. We're perfectly fine changing it so that Mavo can be used for primary-aged children! I have a young daughter myself and would love to build Mavo apps for her when she's a little older 😃 

Any ideas on the new wording?

@karger @DmitrySharabin

This is the dialoog we're talking about:
![image](https://user-images.githubusercontent.com/175836/82160022-5b0bbe00-9860-11ea-849f-98f8acea1779.png)

**Comment by DmitrySharabin** [comment]
Hey there! 👋🏻

I looked through the search results for the “ask for permissions UI” and noticed a rather regular pattern: the “not now” button for the rejection. How about using “No, not now” in our case?

**Comment by LeaVerou** [comment]
The "wtf" in this case means "I don't recognize this site", so we need something to that effect, but shorter. Because every Mavo app shares the same Github OAuth App, this was added as a precaution to prevent bad actors logging people in against their will.

Maybe it should be just "I don't recognize this site"...

**Comment by karger** [comment]
I don't see "wtf" on the screenshot you included.  But in any case, how about making this message/display a customizable part of the UI, which mavo already does plenty?

**Comment by LeaVerou** [comment]
@karger I think it's a good idea to change it anyway. It's on the red button in the screenshot.

**Comment by DmitrySharabin** [comment]
What if we add a short description after the question, like so: If you don’t recognize the site answer “No”? And keep the text on the button short—only `No`.

**Comment by LeaVerou** [comment]
@DmitrySharabin That is a bit of a usability antipattern:

It's basically text that instructs people how to answer a question in a different part of the UI because its options are not descriptive enough. And descriptive text is often skimmed.

So far I'm leaning towards "No, I don't recognize this site".

**Comment by DmitrySharabin** [comment]
> That is a bit of a usability antipattern

Oops. Thanks, Lea. Now I'll be aware of it. 😊

**Comment by karger** [comment]
One thing that concerns me here is that I'm not sure the typical user (or I) really understand what it means to "log in with github" to a site that I do not recognize.  ie, it is not clear what the consequences of that login are.  Does that site get my password (no, that's the point of oauth).  Does the site get access to some of my data on github?  On other sites of logged into with github?  Does github get access to this site?  

I think we might need a "more info" link for people who care...

**Comment by LeaVerou** [comment]
It gets access to anything you grant access to the Mavo app the first time you authorize it (i.e. on the first mavo app that uses Github).
In any case, let's not allow the better to be the enemy of the good. This button needs renaming ASAP.

**Comment by karger** [comment]
for short term, how about "No"?

**Comment by karger** [comment]
> It gets access to anything you grant access to the Mavo app the first time you authorize it (i.e. on the first mavo app that uses Github).

Does "the mavo app" mean "this particular mavo app" or "all data ever created with any mavo app that stores on github"?

**Comment by LeaVerou** [comment]
The latter. The former wouldn't be a problem.

**Comment by raytran** [comment]
Is there any reason that this has not been changed yet? The popup still says "wtf??"

**Comment by LeaVerou** [comment]
We never reached consensus on the alternative wording and OP never replied.

**Comment by twistylittleworkshop** [comment]
Apologies - I wasn't aware anyone was waiting on me. Last I saw on this, developers were discussing between various patterns or anti-patterns for the wording, and I was lurking with interest.
If you are after my personal take, as long as it is appropriate to an all-age audience, as I feel the software is, then whatever choice is fine - a simple "Yes/No" could suffice until dev consensus on the optimum wording could be reached? 
    On Monday, 2 November 2020, 15:26:20 GMT, Lea Verou <notifications@github.com> wrote:  
 
 


We never reached consensus on the alternative wording and OP never replied.

**Comment by LeaVerou** [comment]
Cool, let's go with "No, I don’t recognize this site". @raytran wanna work on this? It should be a very quick fix.

**Comment by raytran** [comment]
Sure

**Comment by LeaVerou** [comment]
Cool! The repo is https://github.com/mavoweb/auth.mavo.io (private) Let me know if you don't have access!
